include/vctr/Expressions/Core/Mean.h

```diff
@@ -0,0 +1,243 @@
+/*
+  ==============================================================================
+    DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+
+    Copyright 2023 by sonible GmbH.
+
+    This file is part of VCTR - Versatile Container Templates Reconceptualized.
+
+    VCTR is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License version 3
+    only, as published by the Free Software Foundation.
+
+    VCTR is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License version 3 for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    version 3 along with VCTR.  If not, see <https://www.gnu.org/licenses/>.
+  ==============================================================================
+*/
+
+namespace vctr::Expressions
+{
+
+template <size_t extent, class SrcType>
+requires is::number<ValueType<SrcType>>
+class Mean : public ExpressionTemplateBase
+{
+public:
+    using value_type = ValueType<SrcType>;
+
+    VCTR_COMMON_UNARY_EXPRESSION_MEMBERS (Mean)
+
+    static constexpr value_type reductionResultInitValue = 0;
+
+    VCTR_FORCEDINLINE void reduceElementWise (value_type& result, size_t i) const
+    {
+        result += src[i];
+    }
+
+    VCTR_FORCEDINLINE value_type reduceVectorOp() const
+    requires is::suitableForAccelerateRealFloatVectorReductionOp<SrcType, value_type, detail::dontPreferIfIppAndAccelerateAreAvailable>
+    {
+        return Expression::Accelerate::mean (src.data(), size());
+    };
+
+    VCTR_FORCEDINLINE value_type reduceVectorOp() const
+    requires is::suitableForIppRealOrComplexFloatVectorReductionOp<SrcType, value_type, detail::preferIfIppAndAccelerateAreAvailable>
+    {
+        return Expression::IPP::mean (src.data(), sizeToInt (size()));
+    };
```

Comment:
Is it on puprose that some of the reduce methods compute the sum and others compute the mean?

Reply:
Is finalizeReduction not called in this case?

Reply:
I see! Admittedly, didn't know how exactly it works!